server/src/main/java/org/elasticsearch/index/mapper/MappingLookup.java

```diff
@@ -257,8 +258,8 @@ public PostingsFormat getPostingsFormat(String field) {
      */
     public KnnVectorsFormat getKnnVectorsFormatForField(String field) {
         Mapper fieldMapper = fieldMappers.get(field);
-        if (fieldMapper instanceof PerFieldKnnVectorsFormatFieldMapper) {
-            return ((PerFieldKnnVectorsFormatFieldMapper) fieldMapper).getKnnVectorsFormatForField();
+        if (fieldMapper instanceof DenseVectorFieldMapper) {
```

Comment:
drive-by comment, that has little to do with this change specifically, as this logic was already here. Do we need an additional public method in MappingLookup for this? Would it make sense to streamline this into the Mapper base class to be overridden by DenseVectorFieldMapper?

Reply:
Yes, I'm not sure why I thought we needed this! I can remove that `MappingLookup` method now. I am not sure if we want a new top-level method `Mapper#getKnnVectorsFormatForField`, as opposed just doing an `instanceof` check for `DenseVectorFieldMapper`? My inspiration for this change was `CompletionFieldMapper`, which has a special package-private method to define `PostingsFormat`.

Reply:
thanks for the background Alan, makes sense to me as well.